### DISCUSSION TITLE
Synthax highlighting ?

### DISCUSSION THREAD

**Discussion by NicTanghe** [wooorm/markdown-rs#145] [Q&A]
Hello so the main reason i wanted markdown was for syntax highlighted codebloks.

But it seems that common-mark leaves that part to the renderer.
Does anyone know how to add this ?

Is it just CSS ?
how do i do this ?

**Comment by wooorm** [comment]
Syntax highlighting is not something markdown does.

That would be done by plugins: https://github.com/wooorm/markdown-rs/issues/32.

You should be able to use most syntax highlighters. Such as this one by me: https://github.com/wooorm/starry-night#example-using-starry-night-on-the-client (deep link to example of client-side highlighting).

**Reply by NicTanghe** [reply]
I rolled into using syncamore whatever for highlighting with an api. splitting the text and rejoining it being a bit of a pain in leptos but I hope I`ll get there.
(im doing this becouse i want vex highlighting and its not a defoult one)  
also no offense but i feel like using random peoples highlighters wil get me ending up in dependency hell at some point.

There is also a hard nono on anny js in my code.
